### PR TITLE
zos: don't error when killing a zombie process

### PR DESCRIPTION
**Pull request by zsw007**
On z/OS, EPERM is returned if the process being killed is a zombie. However, this shouldn't be treated as an error in `uv_kill()` since it's already terminated, so return 0 instead.

### CODE REVIEW

**Comment by zsw007** [comment]
Draft: https://github.ibm.com/open-z/libuv/pull/16

Upstream: https://github.com/libuv/libuv/pull/3620

**Comment by vtjnash** [comment]
This seems consistent with the errata section of the relevant standard noting that some known implementations are not entirely standards compliant in this manner https://pubs.opengroup.org/onlinepubs/9699919799/functions/kill.html despite claiming standards compliance at https://www.ibm.com/docs/en/zos/2.1.0?topic=functions-kill-send-signal-process

File: src/unix/process.c [hunk 1]
```diff
@@ -1063,9 +1063,14 @@ int uv_process_kill(uv_process_t* process, int signum) {
 
 
 int uv_kill(int pid, int signum) {
-  if (kill(pid, signum))
+  if (kill(pid, signum)) {
+#if defined(__MVS__)
+    /* EPERM is returned if the process is a zombie. */
+    if (errno == EPERM && getpgid(pid) == getpgid(0))
```
Comment (vtjnash) [inline src/unix/process.c:1069]: Would it be more accurate to test exactly if the process is still alive at this point?
```suggestion
    if (errno == EPERM && waitid(P_PID, pid, &infop, WNOHANG | WNOWAIT) == 0)
```

Comment (zsw007) [reply src/unix/process.c]: Makes sense, I have adjusted the code accordingly.

**Review by vtjnash — APPROVED** [review]
Still SGTM

File: src/unix/process.c [hunk 2]
```diff
@@ -1063,9 +1063,15 @@ int uv_process_kill(uv_process_t* process, int signum) {
 
 
 int uv_kill(int pid, int signum) {
-  if (kill(pid, signum))
+  if (kill(pid, signum)) {
+#if defined(__MVS__)
+    /* EPERM is returned if the process is a zombie. */
+    siginfo_t infop;
+    if (errno == EPERM && waitid(P_PID, pid, &infop, WNOHANG | WNOWAIT) == 0)
```
[the remark this answers is not in the document]
Comment (zsw007) [reply src/unix/process.c]: Changes made

**Comment by vtjnash** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/2248/

**Pull request merged**